raiden-ts/src/services/actions.ts

```diff
@@ -45,3 +45,26 @@ export interface iouClear extends ActionType<typeof iouClear> {}
 
 export const udcDeposited = createAction('udc/deposited', UInt(32));
 export interface udcDeposited extends ActionType<typeof udcDeposited> {}
+
+export const udcWithdraw = createAsyncAction(
+  t.partial({}),
+  'udc/withdraw/request',
+  'udc/withdraw/success',
+  'udc/withdraw/failure',
+  t.type({ amount: UInt(32) }),
+  t.type({ txHash: Hash, amount: UInt(32) }),
+);
+
+export namespace udcWithdraw {
+  export interface request extends ActionType<typeof udcWithdraw.request> {}
+  export interface success extends ActionType<typeof udcWithdraw.success> {}
+  export interface failure extends ActionType<typeof udcWithdraw.failure> {}
+}
+
+export const udcWithdrawPlanned = createAction(
+  'udc/withdraw/planned',
+  t.type({ amount: UInt(32), block: UInt(32) }),
+  t.partial({ txHash: Hash }),
```

Comment:
The meta here is the txHash, right? This doesn't seem descriptive enough as an 'identifier' of the planning, since it would only be available after request is mined (so can't be shared with `udcWithdraw`). Maybe you could have `amount` as a general `UdcWithdrawId` codec for `meta`, and use it for `udcWithdraw` and `udcWithdrawPlanned`, and the rest of the parameters in the `payload`. Also, `block` here is a small number, it'd be more consistent with the rest of the SDK to have it as `t.number`, and maybe call it `txBlock`, for consistency, even if it isn't a confirmable action (but could be as well).

Reply:
Ok, can you provide an explanation on when to put something in meta and when in the payload? We had that discussion in the past but a lot of time has passed.

Reply:
`meta` is intended to be data that 'uniquely' identifies actions of a family between different 'instances'. E.g. when a channel action go through (request, event, etc), the meta is `{ tokenNetwork, partner }`, because this uniquely identifies on which channel this action acts. Other actions of the same type but on different channel would not conflict even if happening in parallel. So, for your udcWithdraw actions, which data/property identifies a time series of actions which belong to a specific request? This goes on meta, the rest is specific to each individual action and go on payload (including confirmable params, etc)

Reply:
Forget about `txBlock` on planned action, just saw it's not the tx block, but the withdraw block... it still can be a number though =)